### DISCUSSION TITLE
Swapout check add guardian exclusions

### DISCUSSION THREAD

**Discussion by DaVidoSS0** [EDCD/EDDI#2533] [Needs thought]
Hi. The swapout check can go violent if you have some guardian modules. I get lots of messges about swapping possibility for reduce cost. This can be painfull if you docking oft.

**Comment by nepomuk16321** [comment]
If the message comes too often, you could try to counteract it by changing the calculation values. Or prevent the call of the script in general. For this you have to search the whole character after calling the script and change the call as shown below.
Search for: {F(Swapout check)}
Change to: {_ F(Swapout check) _}
Content of {_ _} will be recognized as text/note only.

**Comment by DaVidoSS0** [comment]
Thanks for reply. The swapout check can't be dissabled from ui checkbox.Is grayed out. By editing the script you mean this ? :
`{_ Function to find a given module in outfitting}
{set outfitting(module) to:
    {for outfittingmodule in station.outfitting:
        {if outfittingmodule.EDName = module.EDName:
            {return outfittingmodule}
        }
    }
}

{_ Function to report if a module is cheaper in outfitting that on the ship}
{set reportIfCheaper(shipmodule) to:
    {_ Only consider if this has no modifications}
    {if !shipmodule.modified:
        {set outfittingmodule to outfitting(shipmodule)}
        {if outfittingmodule:
            {if outfittingmodule.price > 0 && outfittingmodule.price < shipmodule.price:
                {set saving to shipmodule.price - outfittingmodule.price}
                {_ Only bother if it's a relatively good saving... more than 2% of the ship value or more than 0.1% of the commander's credit balance}
                {if saving / ship.value > 0.02 || (cmdr.credits > 0 && saving / cmdr.credits > 0.001):
                    You could reduce your re-buy by {Humanise(round(saving * cmdr.insurance))} credits by swapping out your {shipmodule.name} here.
                }
            }
        }
    }
}

{_reportIfCheaper(ship.powerplant)}
{_reportIfCheaper(ship.thrusters)}
{_reportIfCheaper(ship.frameshiftdrive)}
{_reportIfCheaper(ship.lifesupport)}
{_reportIfCheaper(ship.powerdistributor)}
{_reportIfCheaper(ship.sensors)}
{for hardpoint in ship.hardpoints:
    {if hardpoint.module:
        {_reportIfCheaper(hardpoint.module)}
    }
}
{for compartment in ship.compartments:
    {if compartment.module:
        {_reportIfCheaper(compartment.module)}
    }
}`

**Comment by nepomuk16321** [comment]
That's right, you can't disable the function via the UI. You have to search all scripts after calling the "Swapout check" script. The script is called like this {F("Swapout check")}. If you add a "_" to the braces (so {_ text _}, the content will be interpreted as text and not called. I'll have a look in my character how often "Swapout check" is called.

In my character, the script is called in two other scripts, and those are "Market information updated" and "Stored modules". In these two scripts I would have to change the notation as described above so that the "Swapout check" script is not called.
Ok, I see a mismatch in the "Market information updated" script. Here the variable "swapoucheck" is filled with F("Swapout check"). Here you would have to change the script so that the script is not called >> {_ set swapoutcheck to F("Swapout check") _}

Do you use the original EDDI character? In this I can also look again. Do you have an "own" character, you have to search at your place. I use "notepad++" for it.

Ok, here is the original EDDI script "Market information updated". In line 32 change {swapoutcheck} to {_ swapoutcheck _} or just delete this entry.

```
{_ Find out in advance what we have to say }

{if len(event.updates) > 1:
    {set type to "Station services"}
    {if ship.role = "Multipurpose" || ship.role = "Trading":
        {set purchasecheck to F("Commodity purchase check")}
    }
    {set salecheck to F("Commodity sale check")}
    {set swapoutcheck to F("Swapout check")}
|elif event.updates[0] = "market":
    {set type to OneOf("Market", "Local market")}
    {if ship.role = "Multipurpose" || ship.role = "Trading":
        {set purchasecheck to F("Commodity purchase check")}
    }
    {set salecheck to F("Commodity sale check")}
|elif event.updates[0] = "outfitting":
    {set type to "Station outfitting"}
    {set swapoutcheck to F("Swapout check")}
|elif event.updates[0] = "shipyard":
    {set type to "Station shipyard"}
}

{Pause(500)}
{if purchasecheck || salecheck || swapoutcheck:
    {type}
    {OneOf("data", "information", "price")}
    {OneOf("update", "refresh", "changes")}
    {OneOf("received", "obtained")}.

    {salecheck}
    {purchasecheck}
    {_ swapoutcheck _}             {_ or delete this line _}
}
{F("Limpet check")}

```

**Comment by DaVidoSS0** [comment]
Thank you.
I am using a copy of original scripts, created by copy personality function.
There must be a typo with limpets check function as well. I do not need a limpets reminder when i click shipyard.

**Comment by Tkael** [comment]
Hi there.

I'll update the `Swapout Check` script to exclude guardian modules by revising lines 12-13 as follows:
```
    {_ Only consider if this has no modifications and is not a specialized Guardian module}
    {if !shipmodule.modified && find(module.edname, "Guardian") = -1:
```

I'll also shift the Limpet Check script from being triggered by `Market Information Updated` to `Docked`.
